t/09_qesapdeployment.t

```diff
@@ -789,4 +789,95 @@ subtest '[qesap_prepare_env::qesap_create_aws_config] not solved template and va
     like $contents[0], qr/region = eu-central-1/, "Expected region eu-central-1 is in the config file";
 };
 
+subtest '[qesap_is_job_finished]' => sub {
+    my $qesap = Test::MockModule->new('qesapdeployment', no_auto => 1);
+    my @results = ();
+    $qesap->redefine(script_output => sub {
+            if ($_[0] =~ /100000/) { return "not json"; }
+            if ($_[0] =~ /200000/) { return "{\"job\":{\"state\":\"donaldduck\"}}"; }
+            if ($_[0] =~ /300000/) { return "{\"job\":{\"state\":\"running\"}}"; }
+    });
+
+    $qesap->redefine(get_required_var => sub { return ''; });
+    $qesap->redefine(record_info => sub { note(join(' ', 'RECORD_INFO -->', @_)); });
+
+    push @results, qesap_is_job_finished(100000);
+    push @results, qesap_is_job_finished(200000);
+    push @results, qesap_is_job_finished(300000);
+
+
+    ok($results[0] == 0, "Consider 'running' state if the openqa job status response isn't JSON");
+    ok($results[1] == 1, "Considered 'finished' state if the openqa job status response exists and isn't 'running'");
+    ok($results[2] == 2, "Consider 'running' if the openqa job status response is 'running'");
+};
+
+subtest '[qesap_az_get_active_peerings]' => sub {
+    my $qesap = Test::MockModule->new('qesapdeployment', no_auto => 1);
+    my %results;
+    my @croak_calls;
+
+    $qesap->redefine(script_output => sub {
+            if ($_[0] =~ /myresourcegroup/ && $_[0] =~ /myvnetname/) {
+                return "vnet123456-vnet-other\nvnet789012-vnet-other\nvnet-nojobid-vnet-other";
+            }
+    });
+
+    $qesap->redefine(croak => sub {
+            push @croak_calls, join(' ', @_);
+            die $_[0];
+    });
+
+    # Test missing arguments
+    eval { qesap_az_get_active_peerings(rg => 'myresourcegroup'); };
+    ok($croak_calls[0] =~ /Missing mandatory vnet argument/, "Croak on missing vnet argument");
+
+    eval { qesap_az_get_active_peerings(vnet => 'myvnetname'); };
+    ok($croak_calls[1] =~ /Missing mandatory rg argument/, "Croak on missing rg argument");
+
+    # Test correct id extraction
+    %results = qesap_az_get_active_peerings(rg => 'myresourcegroup', vnet => 'myvnetname');
+    my %expected_results = (
+        "vnet123456-vnet-other" => 123456,
+        "vnet789012-vnet-other" => 789012
+    );
+
+    foreach my $key (keys %expected_results) {
+        ok($results{$key} == $expected_results{$key}, "Correct job id extracted for vnet name $key");
+    }
+
+    # Test incorrect job ID
+    ok(!exists $results{"vnet-nojobid-vnet-other"}, "No job id extracted for vnet name without a valid job id");
+};
+
+subtest '[qesap_az_clean_old_peerings]' => sub {
```

Comment:
This is a test about a qesap_az function (qesap function composing az commands): please move this test to t/15_qesap_azure.t